internal/api/handler/create_stage_v1alpha1.go

```diff
@@ -25,19 +26,28 @@ func CreateStageV1Alpha1(
 		ctx context.Context,
 		req *connect.Request[svcv1alpha1.CreateStageRequest],
 	) (*connect.Response[svcv1alpha1.CreateStageResponse], error) {
-		if req.Msg.GetProject() == "" {
-			return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("project should not be empty"))
-		}
-		if req.Msg.GetName() == "" {
-			return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("name should not be empty"))
-		}
-
-		stage := kubev1alpha1.Stage{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: req.Msg.GetProject(),
-				Name:      req.Msg.GetName(),
-			},
-			Spec: fromStageSpecProto(req.Msg.GetSpec()),
+		var stage kubev1alpha1.Stage
+		switch {
+		case req.Msg.GetYaml() != "":
+			if err := yaml.Unmarshal([]byte(req.Msg.GetYaml()), &stage); err != nil {
+				return nil, connect.NewError(connect.CodeInvalidArgument, errors.Wrap(err, "invalid yaml"))
+			}
+		case req.Msg.GetTyped() != nil:
+			if req.Msg.GetTyped().GetProject() == "" {
+				return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("project should not be empty"))
+			}
+			if req.Msg.GetTyped().GetName() == "" {
+				return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("name should not be empty"))
```

Comment:
Perhaps this edge case is covered by `kc.Create`, but should we ensure that the Project (namespace) exists?

Actually, since we are now using a label to distinguish Kargo projects from normal namespaces, we probably need an explicit check here that the Kargo project specified in this create call exists.